### PR TITLE
Snapshot test suggestion

### PR DESCRIPTION
**Pull request by grillorafael**
Hey @sksamuel.

Just a test change suggestion. Would like to know what do you think.

Start using snapshot matcher in favor of match json resource and simple json string match.

I put a commend on the test that actually is using this matcher.

You can find more info here: https://github.com/commodityvectors/scalatest-snapshot-matchers

### CODE REVIEW

File: elastic4s-core/src/test/scala/com/sksamuel/elastic4s/mappings/MappingDefinitionDslTest.scala [hunk 1]
```diff
@@ -64,31 +71,32 @@ class MappingDefinitionDslTest extends WordSpec with Matchers with JsonSugar wit
...
-    "include timestamp store field" in {
+    "include timestamp store field" in { implicit test =>
       val req = createIndex("docsAndTags").mappings(
         mapping("foo") timestamp {
           timestamp(true) path "mypath" store true
         }
       )
-      CreateIndexContentBuilder(req).string() should matchJsonResource("/json/mappings/timestamp_store.json")
+
+      CreateIndexContentBuilder(req) should matchSnapshot[XContentBuilder]()
```
Comment (grillorafael) [inline elastic4s-core/src/test/scala/com/sksamuel/elastic4s/mappings/MappingDefinitionDslTest.scala]: This one

**Comment by sksamuel** [comment]
Is this your project? If so, I'd consider adding some explanation to the readme on what it does. I've read the front page and I'm not sure what a snapshot is and why it would be useful to use it.

**Comment by grillorafael** [comment]
Yes it's mine. It's a scalatest implementation of Jest's snapshot testing (https://www.youtube.com/watch?v=HAuXJVI_bUs). Basically what it does is it facilitates matching by automatically generating your match file and when there is a mismatch, showing the diff and prompting for updates.

You can run the change test changing the json content to see how it works as well. It might be helpful for testing json like strings like this. We are currently using in my current job's projects and it is working quite well for us.

That is just a suggestion in case you think there is a fit for elastic4s.

I'll try to improve the readme as well

**Comment by grillorafael** [comment]
I'll close this as it isn't hugely important and will not bring much improvements overall

**Pull request closed without merging**